test/xpath/test_base.rb

```diff
@@ -1193,6 +1193,16 @@ def test_namespaces_0
       assert_equal( 1,  XPath.match( d, "//x:*" ).size )
     end
 
+    def test_namespaces_cache
+      doc = Document.new "<a xmlns='1'><b/></a>"
+      assert_equal("<b/>", XPath.first( doc, "//b[namespace-uri()='1']" ).to_s)
+      assert_nil(XPath.first( doc, "//b[namespace-uri()='']" ))
+
+      doc.root.delete_namespace
+      assert_nil(XPath.first( doc, "//b[namespace-uri()='1']" ))
+      assert_equal("<b/>", XPath.first( doc, "//b[namespace-uri()='']" ).to_s)
```

Comment:
(Do we need the `"//b[namespace-uri()='1']"` case?)

```suggestion
      assert_nil(XPath.first(doc, "//b[namespace-uri()='1']"))
      assert_equal("<b/>", XPath.first(doc, "//b[namespace-uri()='']").to_s)
```

Reply:
> (Do we need the `"//b[namespace-uri()='1']"` case?)

I think this is necessary to see changes in cache status.